app/routes/facebook.js

```diff
@@ -0,0 +1,48 @@
+import Route from '@ember/routing/route';
+
+export default Route.extend({
+  queryParams: {
+    code     : '',
+    provider : ''
+  },
+  model(queryParams) {
+    this.get('loader').post(`/auth/oauth/login/${ queryParams.provider }?code=${ queryParams.code }`)
+      .then(response => {
+        let credentials = {
+              identification : response.email,
+              password       : response.oauth_hash
+            },
+            authenticator = 'authenticator:jwt';
+
+        this.get('session')
+          .authenticate(authenticator, credentials)
+          .then(async() => {
+            const tokenPayload = this.get('authManager').getTokenPayload();
+            if (tokenPayload) {
+              this.get('authManager').persistCurrentUser(
+                await this.get('store').findRecord('user', tokenPayload.identity)
+              );
+            }
+            window.location.replace('/');
```

Comment:
That is not how you transition in an ember app. Use the native methods of ‘transitionTo’ and ‘transitionToRoute’ instead.